test/e2e/specs/onboarding/signup__free.ts

```diff
@@ -74,7 +74,16 @@ describe( DataHelper.createSuiteTitle( 'Signup: WordPress.com Free' ), function
 
 		it( 'Select "write" path', async function () {
 			startSiteFlow = new StartSiteFlow( page );
-			await startSiteFlow.clickButton( 'Start writing' );
+
+			await page.waitForLoadState( 'networkidle' );
+			const isGoalsStep = await page.isVisible( ':has-text("What are your goals")' );
```

Comment:
It is strongly suggested not to use `isVisible` as it will immediately return a boolean value. This means that any delay or async loading can set `isGoalsStep` down the wrong path.

It would be more preferable to use the `locator.waitFor` API, either on its own or in conjunction with the `page.waitForLoadState` call.

If you are certain that the target text will be loaded once `networkidle` is fired, you could try this:

Suggested change:

```typescript
await page.waitForLoadState( 'networkidle' );
const goalsLocator = page.locator( ':has-text("What are your goals")' );
if (await goalsLocator.count() > 0) {
   // true case
} else {
   // false case
}

Alternatively, it is possible to [wait on multiple selectors](https://playwright.dev/docs/selectors#css-selector-list), if it makes sense.

Reply:
Additionally, if making changes it is preferable to make the change in the `StartSiteFlow` page object itself, rather than patching it in the spec, as the latter approach results in repeated/duplicated code.

Suggestion: add a method called `startSiteFlow.selectPath`, which will take a parameter _path_ to denote which path the user wants to click on.

If the param is `write`, enter the logic in Lines 81-86.

That way, the resulting spec calls `startSiteFlow.selectPath( 'write' )`

Feel free to let me know if you'd like me to show an example of what I mean!

Reply:
Ah, I see this is already merged @yansern and you're on AFK! I'll take a look at making these improvements once the follow-up PR #64877 is merged. I'd love to review PRs related to the framework or e2e tests before merge next time!

Reply:
I'm working on this feedback in https://github.com/Automattic/wp-calypso/pull/64877

Reply:
Thanks @worldomonation & @roo2 for reviewing & following it up. Yes, 2 weeks ago I merged this in a rush to get it fixed (speed over quality) as my AFK was coming soon. :)